packages/react/src/components/Table/TableViewDropdown/TableViewDropdown.jsx

```diff
@@ -71,121 +71,127 @@ const defaultProps = {
-        : [saveAsNewItem, manageViewsItem];
+    const allItems = useMemo(() => {
+      const saveAsNewItem = {
+        id: 'save-new-view',
+        text: i18n.saveAsNewView,
+        customAction: onSaveAsNewView,
+      };
+      const saveItem = {
+        id: 'save-changes',
+        text: i18n.saveChanges,
+        customAction: onSaveChanges,
+      };
+      const manageViewsItem = {
+        id: 'manage-views',
+        text: i18n.manageViews,
+        customAction: onManageViews,
+        icon: Settings16,
+      };
+      // Save changes button show only appear if the view has been edited and the current view is not 'View all'
+      // 'View all' is equivalent to a "default view", which would not be able to get re-saved. The user should supply
+      // their own default views that can be changed if they would like that functionality
+      const dialogItems =
+        selectedViewEdited && selectedViewId && selectedViewId !== 'view-all'
+          ? [saveAsNewItem, saveItem, manageViewsItem]
+          : [saveAsNewItem, manageViewsItem];
 
-    // move the action / dialog items to the top so that they are always easily accessible in the case that there are
-    // many views. The user would need to scroll all the way to the bottom to find the actions
-    return [...dialogItems, viewAllItem, ...views];
-  }, [
-    i18n.saveAsNewView,
-    i18n.saveChanges,
-    i18n.manageViews,
-    onSaveAsNewView,
-    onSaveChanges,
-    onManageViews,
-    selectedViewEdited,
-    selectedViewId,
-    viewAllItem,
-    views,
-  ]);
+      // move the action / dialog items to the top so that they are always easily accessible in the case that there are
+      // many views. The user would need to scroll all the way to the bottom to find the actions
+      return [...dialogItems, viewAllItem, ...views];
+    }, [
+      i18n.saveAsNewView,
+      i18n.saveChanges,
+      i18n.manageViews,
+      onSaveAsNewView,
+      onSaveChanges,
+      onManageViews,
+      selectedViewEdited,
+      selectedViewId,
+      viewAllItem,
+      views,
+    ]);
 
-  const mySelectedItem = allItems.find((item) => item.id === selectedViewId) || viewAllItem;
-  const MyDropDown = overrides?.dropdown?.component || Dropdown;
-  const MyTableViewDropDownItem = overrides?.dropdownItem?.component || TableViewDropdownItem;
-  const [containerSize, containerRef] = useSizeObserver({ initialWidth: 200 });
-  const [dropdownRef, updateTitle] = useDropdownTitleFixer();
+    const mySelectedItem = allItems.find((item) => item.id === selectedViewId) || viewAllItem;
+    const MyDropDown = overrides?.dropdown?.component || Dropdown;
+    const MyTableViewDropDownItem = overrides?.dropdownItem?.component || TableViewDropdownItem;
+    const [containerSize, containerRef] = useSizeObserver({ initialWidth: 200 });
+    // destructure the array as an object by index
+    const { 1: updateTitle } = useDropdownTitleFixer(ref?.current);
```

Comment:
I ended up changing this in a later commit. The useDropdownTitleFixes doesn't take a param. It returns the ref from the hook. I initially added it because I was getting a "forwardRef" error, but I realized that was because of an override component being used in the test/story that was just a function component. So passing the ref isn't needed after I fixed that with forwardRef.